src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java

```diff
@@ -34,28 +34,40 @@
 import java.net.URLEncoder;
 import java.io.UnsupportedEncodingException;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
 /** Servlet to take in dish name and return bulk recipe infomation */
 @WebServlet("/dishNutrition")
 public class SpoonacularNutritionServlet extends HttpServlet {
   private static String spoonacularPrefix = "https://api.spoonacular.com/recipes/guessNutrition";
-  private static String spoonacularAPIKey = "cd2269d31cb94065ad1e73ce292374a5";
 
   /** Takes in dishName parameter and uses spoonacular's guess nutrition to return nutrition information about the dish*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String title = request.getParameter("dishName");
-    System.out.println(title);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+      new Query("apiKey")
+        .setFilter(new Query.FilterPredicate("keyName", Query.FilterOperator.EQUAL, "spoonacular"));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    String spoonacularKey = (String) entity.getProperty("key");
```

Comment:
Consider wrapping this part into a helper function that can be shared by multiple components.